crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -161,14 +161,51 @@ impl Ethereum {
             .map_err(Into::into)
     }
 
-    pub async fn decimals(&self, address: eth::TokenAddress) -> Result<u8, Error> {
-        let erc20 = self.contract_at::<contracts::ERC20>(address.0);
-        erc20.methods().decimals().call().await.map_err(Into::into)
+    /// Returns the token's decimals. Returns `None` if the token does not
+    /// implement this optional method.
+    pub async fn decimals(&self, token: eth::TokenAddress) -> Result<Option<u8>, Error> {
+        let erc20 = self.contract_at::<contracts::ERC20>(token.0);
```

Comment:
What I meant in my other comment was to use `contract_at` in the `tokens::Fetcher` code, not necessarily to add new ERC-20 specific methods to the `Ethereum` type (which feels a bit weird, as you pointed out before).

Reply:
Ahh, okay. Misunderstood you.
Fixed in: [d2bbf31](https://github.com/cowprotocol/services/pull/1772/commits/d2bbf312faeab8a43cb562607bbf219f5f9c8152)